src/Button/ButtonBase.tsx

```diff
@@ -62,40 +66,60 @@ const ButtonBase = forwardRef(
     }
 
     return (
-      <StyledButton
-        as={Component}
-        sx={sxStyles}
-        {...rest}
-        ref={innerRef}
-        data-block={block ? 'block' : null}
-        data-size={size === 'small' || size === 'large' ? size : undefined}
-        data-no-visuals={!LeadingVisual && !TrailingVisual && !TrailingAction ? true : undefined}
-      >
-        {Icon ? (
-          <Icon />
-        ) : (
-          <>
-            <Box as="span" data-component="buttonContent" sx={getAlignContentSize(alignContent)}>
-              {LeadingVisual && (
-                <Box as="span" data-component="leadingVisual" sx={{...iconWrapStyles}}>
-                  <LeadingVisual />
-                </Box>
-              )}
-              {children && <span data-component="text">{children}</span>}
-              {TrailingVisual && (
-                <Box as="span" data-component="trailingVisual" sx={{...iconWrapStyles}}>
-                  <TrailingVisual />
+      <>
+        <StyledButton
+          as={Component}
+          sx={sxStyles}
+          {...rest}
+          ref={innerRef}
+          data-block={block ? 'block' : null}
+          data-size={size === 'small' || size === 'large' ? size : undefined}
+          data-no-visuals={!LeadingVisual && !TrailingVisual && !TrailingAction ? true : undefined}
+          aria-disabled={loading ? true : undefined}
+          aria-describedby={loading ? 'loading-message' : undefined}
```

Comment:
Should this be?
```suggestion
          aria-describedby={loading ? loadingMessage : undefined}
```

Reply:
I'm pretty sure `aria-describedby` is meant to reference the `id` of the message/string, not actually contain the string.

Reply:
Ohhh sorry yes 100% 😄